python/lsst/daf/butler/formatters/parquet.py

```diff
@@ -290,6 +292,38 @@ def arrow_to_numpy_dict(arrow_table: pa.Table) -> dict[str, np.ndarray]:
     return numpy_dict
 
 
+def _numpy_dict_to_numpy(numpy_dict: dict[str, np.ndarray]) -> np.ndarray:
+    """Convert a dict of numpy arrays to a structured numpy array.
+
+    Parameters
+    ----------
+    numpy_dict : `dict` [`str`, `numpy.ndarray`]
+        Dict with keys as the column names, values as the arrays.
+
+    Returns
+    -------
+    array : `numpy.ndarray` (N,)
+        Numpy array table with N rows and columns names from the dict keys.
+    """
+    return arrow_to_numpy(numpy_dict_to_arrow(numpy_dict))
+
+
+def _numpy_to_numpy_dict(np_array: np.ndarray) -> dict[str, np.ndarray]:
+    """Convert a structured numpy array to a dict of numpy arrays.
+
+    Parameters
+    ----------
+    np_array : `numpy.ndarray`
+        Input numpy array with multiple fields.
+
+    Returns
+    -------
+    numpy_dict : `dict` [`str`, `numpy.ndarray`]
+        Dict with keys as the column names, values as the arrays.
+    """
+    return arrow_to_numpy_dict(numpy_to_arrow(np_array))
```

Comment:
Well, there aren't any caught by the tests!  😄  But the conversions are totally reversible (except to/from pandas because pandas) so it should be fine.  There's also a memory penalty going from a row-based numpy array to a column-based arrow table OR astropy table OR numpy dict.  But there is no copying of the underlying data when converting between these column-based formats.